src/query/api/v1/handler/placement/delete.go

```diff
@@ -126,6 +126,14 @@ func (h *DeleteHandler) ServeHTTP(serviceName string, w http.ResponseWriter, r *
 			return
 		}
 
+		_, ok := curPlacement.Instance(id)
+		if !ok {
+			logger.Info("instance not found in placement", zap.String("instance", id))
+			err := fmt.Errorf("instance %s not found in placement", id)
+			xhttp.Error(w, err, http.StatusNotFound)
```

Comment:
Should this be StatusBadRequest?

Reply:
I wanted NotFound so that it's more clear to clients what went wrong, and that they don't have to retry since the instance they were trying to delete is already deleted